src/main/java/io/supertokens/storageLayer/StorageLayer.java

```diff
@@ -432,64 +437,42 @@ public static TenantIdentifierWithStorageAndUserIdMapping getTenantIdentifierWit
         throw new UnknownUserIdException();
     }
 
-    public static AppIdentifierWithStorageAndUserIdMapping getAppIdentifierWithStorageAndUserIdMappingForUserWithPriorityForTenantStorage(
-            Main main, AppIdentifier appIdentifier, Storage priorityStorage, String userId,
+    public static AppIdentifierWithStorageAndUserIdMapping getAppIdentifierWithStorageAndUserIdMappingForUser(
```

Comment:
restructure this to have three separate if statements and within each do only what you want to do